src/segger/__init__.py

```diff
@@ -1,8 +1,14 @@
-__all__ = ["data", "models", "prediction", "training"]
+import importlib.util
+
+cupy_available = importlib.util.find_spec("cupy") is not None
 
+from segger.data import *
+from segger.models import *
+from segger.training import *
+from segger.validation import *
 
-from .data import *
-from .models import *
-from .prediction import *
-from .training import *
-from .validation import *
+# segger.prediction requires cupy, which is not available in macOS
+if cupy_available:
+    from segger.prediction import *
+
+__all__ = ["data", "models", "prediction", "training"]
```

Comment:
On macOS, `cupy` is not available. `segger.prediction` tries to import `cupy` leading to an import error.

Reply:
approved.